File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp

 // Definition for a binary tree node.
//   struct TreeNode {
//       int val;
//       TreeNode *left;
//       TreeNode *right;

//       TreeNode() : val(0), left(nullptr), right(nullptr) {}
//       TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
//       TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
//   };

class Solution {
public:
    vector<int>ans;
    void bfs(TreeNode* node){
        queue<TreeNode*>que;
        que.push(node);
        int level = 0;
        int l = 1;
        while(que.size() != 0){
            TreeNode* curr = que.front();
            que.pop();
            l--;

            if(curr->left){
                que.push(curr->left);
            }
            if(curr->right){
                que.push(curr->right);
            }

            if(l == 0 && que.size()){
                l = que.size();
                int val = que.back()->val;
                ans.push_back(val);
            }
        }
    }
    
    vector<int> rightSideView(TreeNode* root) {
        if(!root){
            return {};
        }
        ans.push_back(root->val);
        bfs(root);
        return ans;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        def bfs(node):
            que = deque([node])
            l = 1

            while que:
                curr = que.popleft()
                l -= 1
                if curr.left:
                    que.append(curr.left)
                if curr.right:
                    que.append(curr.right)
                
                if not l and que:
                    l = len(que)
                    ans.append(que[-1].val)
        
        if root:
            ans.append(root.val)
            bfs(root)
        
        return ans

        